src/main/java/animals/LanguageRules.java

```diff
@@ -0,0 +1,41 @@
+package animals;
+
+import java.util.Map;
+import java.util.Objects;
+import java.util.ResourceBundle;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import static java.util.stream.Collectors.toUnmodifiableMap;
+
+public abstract class LanguageRules {
+    protected static final Map<String, Pattern> patterns;
+    private static final ResourceBundle rules;
```

Comment:
If you create constants, you need to follow a particular code convention for naming. 
The recommended one is uppercase.